src/Umbraco.Core/Scoping/Scope.cs

```diff
@@ -590,41 +752,138 @@ private void ClearLocks(Guid instanceId)
             {
                 lock (_dictionaryLocker)
                 {
-                    ReadLocks?.Remove(instanceId);
-                    WriteLocks?.Remove(instanceId);
+                    _readLocksDictionary?.Remove(instanceId);
+                    _writeLocksDictionary?.Remove(instanceId);
+
+                    // remove any queued locks for this instance that weren't used.
+                    while (_queuedLocks?.Count > 0)
+                    {
+                        var top = _queuedLocks.PeekStack();
```

Comment:
```suggestion
                        // It's safe to assume that the locks on the top of the stack belong to this instance,
                        // since any child scopes that might have added locks to the stack must be disposed before we try and dispose this isntance.
                        var top = _queuedLocks.PeekStack();
```

A clarifying comment here would be nice, since it's not necessarily apparent at a first glance, I've tried adding something here that hopefully makes some sense 😄